src/main/java/analyzer/exercises/ExerciseFactory.java

```diff
@@ -0,0 +1,11 @@
+package analyzer.exercises;
+
+public interface ExerciseFactory {
+
+    static ExerciseFactory get() {
+        return new ReflectionBasedExerciseFactory();
+    }
+
+    Exercise create(final String slug, final String dir) throws ExerciseNotFoundException;
```

Comment:
Yeah, I am not a big fan of this reflection-based approach.

If we want to get rid of the switch-case, I would prefer to have an ExerciseRegistry that every Exercise class is registered into so that we do not have to keep updating an ever-growing switch-case. The nice thing about that solution is that the exercise class knows which slug it belongs to and our main entry point becomes dead simple. Let me mock that up in a draft PR.

Another option is to use a dependency-injection framework (like Guice), we could probably use bindings instead. That is probably too heavy-handed for our needs.

Reply:
@jmrunkle That is a great idea, I like that a lot better than what has been done here. Could you tag me once that draft PR is online? 

Also, I believe that settles this PR. Should I just go ahead and close it?